lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -3541,6 +3547,30 @@ LogicalResult StmtEmitter::visitSV(GenerateCaseOp op) {
   return success();
 }
 
+LogicalResult StmtEmitter::visitSV(ForOp op) {
+  emitSVAttributes(op);
+  llvm::SmallPtrSet<Operation *, 8> ops;
+  startStatement();
+  auto inductionVarName = op->getAttrOfType<StringAttr>("hw.verilogName");
+  ps << "for (" << PP::ibox2 << "logic ";
+  ps.invokeWithStringOS([&](auto &os) {
+    emitter.emitTypeDims(op.getInductionVar().getType(), op.getLoc(), os);
```

Comment:
Do we ever expect to see `X` or `Z` values in the index? Otherwise you could change `logic` to `bit` such that we only operate on two-valued integers.